File: todoApp/src/app/model/todo.ts
import {Group} from "./group";

export class Todo {
    id: number;
    title: string;
    user: string;
    group: Group;
    createdAt: string;
    done: boolean;
    description: string;

    constructor(id?: number, title?: string, description?: string,  done?: boolean, group?: Group, user?: string, createdAt?: string) {
        this.id = id;
        this.title = title;
        this.user = user;
        this.createdAt = createdAt;
        this.done = done;
        this.description = description;
        this.group = group;
    }

}

File: spring-session/todo-resource/src/main/java/agashchuk/model/Todo.java
package agashchuk.model;

import agashchuk.profileserver.security.User;
import lombok.Data;
import org.springframework.web.multipart.MultipartFile;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.Date;

@Entity(name = "todo")
@Data
public class Todo {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @NotNull
    private String title;
    @ManyToOne
    @JoinColumn(name = "user_id")
    @NotNull
    public User user;
    @NotNull
    public Date createdAt;
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "group_id")
    @NotNull
    public Group group;
    public boolean done;
    public String description;
    @Transient
    private MultipartFile file;

    @Override
    public String toString() {
        return "Book{" +
                "id=" + id +
                '}';
    }
}
